---
title: "Chromosome 20"
author: "Bruce Cochrane"
date: "10/22/2016"
output: html_document
---
```{r}
library(hierfstat)
source("./R/VcfFunctions.R")
```

### Overview

The approach here is to do the following

1.  Download the vcf.gz file for chromosome 20 (chosen arbitrarily) from http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/
2.  Use VCFloci to gt the metadata for the whole thing
3.  Select a sample of N loci
4.  Use the which.loci option of read.vcf to get the raw data from there.

### Read in Data


```{r}
dat20.inf <-VCFloci("./Data/Chromosome20.vcf.gz")
```

### Select subset

```{r}
set.seed(1592)
n <-400
smp <-sample(1:nrow(dat20.inf),n)
dat20.inf.sub <-dat20.inf[smp,]
```

### Read subsetted vcf file 

```{r}
dat20 <-read.vcf("./Data/Chromosome20.vcf.gz",which.loci = smp)

```

### Identify and remove indels

```{r}
trsnps <-dat20.inf.sub$ID[which(dat20.inf.sub$ALT %in% c("A","G","C","T")&dat20.inf.sub$REF %in% c("A","G","C","T"))]

dat.noindel <-dat20[,which(colnames(dat20) %in% trsnps)]

```

Now get rid of monomorphics

```{r}
dat.noindel2 <-monogon(dat.noindel)
```

### Convert to genind

```{r}
dat.gen <-loci2genind(dat.noindel2)
```

### add pop slot

```{r}
samplist <-read.table("./Data/sampleList2.tsv",header=TRUE, sep = "\t") #read sample info file
smplist.sub <-data.frame(samplist[,c(1,4,6)]) # select subject, population and superpopulaton
smps.20 <-rownames(dat.noindel2)
test <-smplist.sub[which(smplist.sub$Sample.name %in% smps.20),]
```

Now, we want to create a vector of populations, in which the names in the genind are associated with pops and super pops, only in order


And then create a vector with the subjects in the data associated with their position in the list of names

```{r}
smp.order <-sapply(smps.20, function (x) which (test$Sample.name==x))

```
Ok.  that gives us a vector of posiotions for the samples as in the genind; now try a simple
rearrangement

```{r}
test.srt <-test[smp.order,]
head(test.srt)
```

Now we can add a vector to the genind object

```{r}
dat.gen@pop <-test.srt$Population.code
```

### Basic Stats

```{r}
bt <-basic.stats(dat.gen)
bt$overall
```

And with those manipulations, we have a believable Fst of 0.08

### PCA


Now, rock on

```{r}
dat.pca <-indpca(dat.gen)
```

```{r}
par(mfrow=c(1,1))
plot(dat.pca,cex=.5,col=as.numeric(test.srt$Superpopulation.code),eigen = FALSE)
```
There is a risk of having an outlier sample skew everything.  

having an outlier skew everything. 
```{r}
chr20 <-list(dat.noindel2,dat.gen,test.srt)
save(chr20,file="./Data/Chromosome20.rda")
```

A little fooling regarding diversity using H0 (see Nei for more)

```{r}
div <-bt$Ho
```

OK  that has one column per population.  Let's average them

```{r}
div.mean <-apply(div,2,mean)
barplot(div.mean,names.arg=names(div.mean),las=2,ylim=c(.025,.05))
```

Need to figure out how to sort these by supecode

```{r}
head(smplist.sub)
```

```{r}
tt <-smplist.sub[,2:3]
unique((tt),1,sort)
```

One possible solution

```{r}
library(data.table)
locs <-unique(as.data.frame(tt))
```

That's a start; still need to sort by population code

```{r}
locs.ord <-locs[with(locs,order(Population.code,Superpopulation.code)),]
```

And now we can do the barplot

```{r}
barplot(div.mean,names.arg=locs.ord[,2],las=2,ylim=c(.025,.05)
```

