---
title: <center>Getting Data</center>
author: <center></center>
date: <center>`r format(Sys.Date(),format="%%B %%d %%Y")`</center>
bibliography: ~/Dropbox/TPG.bib
csl: ~/Dropbox/genetics.csl.txt 
output: 
    html_document:
       css: ~/Dropbox/Chapter.css
       toc: yes
       toc_depth: 5
       code_folding: show
---

### Backgound

The 1000 Genomes Project, phase 3, is the result of analyzing 2504 human genomes from 26 worldwide populations.  Most of the data are available not as complete sequences, but rather as variant information - the differences between an individual's genotype and a standard reference genome.  And of course, individuals possess two copies of each chromosome (except maybe the X) so the issue of phasing is one that must be considered.  Fortunately, as best I can understand, the data have been phased, so we won't worry about that problem.

### File Formats

#### VCF
This is the principle one we will start with, as it can be readily input into R by a function in the population genetics package pegas.  Details of the format can be seen by clicking  [here](http://samtools.github.io/hts-specs/VCFv4.2.pdf); in short it is a complex format that stores variation data and metadata


#### Pedfiles

#### Impute files

### Auxilliary (non-R) programs

#### Haploview

Haploview is a java program that comes with a graphical interface; in general what it does is to infer haplotypes from genotype data.  It requires .ped and .info files as inputs; our primary use for it will be to identify tag SNPs when needed.

#### Vcftools

This is a command-line program that can be downloaded from [https://vcftools.github.io/](https://vcftools.github.io/).  It has a lot of filtering options (if you love the command line), but for our purposes, it can generate an IMPUTE-formatted file, useful in haplotype analysis

### Where to begin

There are two ways one can get 1000G data, either through NCBI or directly from the 1000 genomes web site.  The latter is a bit more intuitive, and it produces compressed VCF files, saving download time and disk space.  It can also output PLINK (ped and map) files, although they can also be generated with vcftools.

The web site for the 1000 genomes browser is [http://browser.1000genomes.org](http://browser.1000genomes.org/index.html), when you land there, you will have a simple search window, into which you can enter a gene name or a chromosomal region (for example Chromosome 2:1000000-2000000).  Note that if you enter a gene name, you will likely 

### Reading multiple populations

In general, we will want to be comparing populations.  There are two ways to accomplish this.  One is by downloading two separate files from the browser, and this is probably the most straightforward.  An alternative is to download them together and then separate them.  The code below accomplishes that for data obtained from the region of the lactase gene (our poster child for selection)
```{r}
dat <-read.vcf("./Data/lct.ceu-yri.vcf.gz")
dat.info <-VCFloci("./Data/lct.ceu-yri.vcf.gz")
```

This returns a list of both the ceu and lct populations, as well as an information file about what is in it.  However, the two populations are not separated.  To do so we first need to load a master list of samples and population IDs and then subset our data:

```{r}
lst <-read.csv("./Data/sampleList.csv",stringsAsFactors = FALSE) #read a static list of population IDs and sample codes
head(lst)
```

And we can then create two separate data sets as follows:

```{r}
nms.ceu <-lst$SampleID[which(lst$Population=="CEU")]
nms.yri <-lst$SampleID[which(lst$Population=="YRI")]
dat.ceu <-dat[which(rownames(dat) %in% nms.ceu),]
dat.yri <-dat[which(rownames(dat) %in% nms.yri),]
```

At this point, we need to eliminate monomorphics and set MAF - I believe there is old code to do this - need to get it imported.

### Get rid of monomorphics

```{r remove monomorphics}
monogon <-function(vfile){
  vfile[, sapply(vfile, nlevels) > 1]
}
  
```



```{r}
dat.ceu.smp <-monogon(dat.ceu)
dat.yri.smp <-monogon(dat.yri)
```

### And set MAF

```{r}
setmaf <-function(vfile, maf=.05){
  vfile.sum <-summary(vfile)

frqs <-sapply(vfile.sum, function(x) min(x$allele)/sum(x$allele)) #Calculate minor allele frequencies

vfile[,which(frqs>=maf)] 
}
  
```

```{r}
dat.ceu.smp <-setmaf(dat.ceu.smp)
dat.yri.smp <-setmaf(dat.yri.smp)
```

### Removing Labels

This is only for cosmetic reasons - to get the snp labels out for plotting

```{r}
lblstrip <-function(vfile){ attr(vfile,"Labels") <-NULL
vfile}
```

### Saving the data

```{r}
lct <-list(ceu.vcf=dat.ceu.smp,yri.vcf=dat.yri.smp)
save(lct,file="./ProcessedData/lct.rda")
```

So flesh in the above, and we have the start of a writeup.  Everything that follows is scratch.

```{r}
ceu.r <-LDscan(dat.ceu.smp)
yri.r <-LDscan(dat.yri.smp)
```

```{r}
LDmap(ceu.r)
```

```{r}
LDmap(yri.r)
```

