---
title: "VCF from multiple chromosomes."
author: "Bruce Cochrane"
date: "10/30/2016"
output: html_document
---
```{r}
library(hierfstat)
source("./R/VcfFunctions.R")
```

### Background

The goal here is to collect a scattering of SNPs.  Ultimately we need to figure out how to make them proportional to length, but for now, we'll just see how it might work, using code from Chromosome20 as a guide.

We'll need to make a function out of this, but for now, we'll run the code twice, once for fourteen and once for 15
```{r}
pth <-"../1000GenomesData/Chromosome" # set base path to files
chr <-"15.vcf.gz" # Substitute number for the one we want
ch15.inf <-VCFloci(paste0(pth,chr))
```

So for now, we'll shoot for 100 loci per chromosome as a test

```{r}
smp <-sample(1:nrow(ch15.inf),100)
ch15.inf.sub <-ch15.inf[smp,]
ch15.dat <-read.vcf(paste0(pth,chr),which.loci=smp)
```

Now edit the code above to get 15, after which we will start processing

### Try combining the two files generated by pegas reads

```{r}
comb.dat <-cbind(ch14.dat,ch15.dat)
comb.dat
```

That worked.  Now to combine the info files.

```{r}
comb.inf <-rbind(ch14.inf.sub,ch15.inf.sub)
```

That all looks good.  Now we'll borrow code from Chromosome20 to see if we can get get to basic.stats and PCA; if so we will save stuff in ./Data and get to real work.

#### Remove indels

```{r}
trsnps <-comb.inf$ID[which(comb.inf$ALT %in% c("A","G","C","T")&comb.inf$REF %in% c("A","G","C","T"))]

dat.noindel <-comb.dat[,which(colnames(comb.dat) %in% trsnps)]

```

### Get rid of monomorphics

```{r}
dat.noindel2 <-monogon(dat.noindel)
```

### Convert to genind

```{r}
dat.gen <-loci2genind(dat.noindel2)
```

### add pop slot

```{r}
samplist <-read.table("./Data/sampleList2.tsv",header=TRUE, sep = "\t") #read sample info file
smplist.sub <-data.frame(samplist[,c(1,4,6)]) # select subject, population and superpopulaton
smps.comb <-rownames(dat.noindel2)
test <-smplist.sub[which(smplist.sub$Sample.name %in% smps.comb),]
```

#### Reorder

```{r}
smp.order <-sapply(smps.comb, function (x) which (test$Sample.name==x))
test.srt <-test[smp.order,]
dat.gen@pop <-test.srt$Population.code
```


## Basic Statistics

```{r}
bt <-basic.stats(dat.gen)
bt$overall
```

Looks good, with Fst up to 0.1

```{r}
comb.pca <-indpca(dat.gen)
```

And plot

```{r}
par(mfrow=c(1,1))
plot(comb.pca,cex=.5,col=as.numeric(test.srt$Superpopulation.code),eigen = FALSE)
```

OK.  This is interesting.  We do have the outlier problem.  I think the course of action will be

1.  Start fresh with a set seed
2.  functionalize some of the above.


But before doing that, I'm curious about something - can we get some idea of chromosome size, so that when we get this all done, we can try to get a working file of maybe 1000 loci that can be subsetted?

```{r}
colnames(comb.inf)
```

```{r}
min(ch14.inf$POS)
range(ch15.inf$POS)
```

Probably the easiest thing would be to find a list of human chromosome lengths and use that for proportioning.  Enough for now.  The basic concept is sound.